docs/user/pings/testing-custom-pings.md

```diff
@@ -0,0 +1,121 @@
+# Unit testing Glean custom pings
+
+Applications defining [custom pings](custom.md) can use use the strategy defined in this document to test these pings in unit tests.
+
+## General testing strategy
+
+The schedule of custom pings depends on the specific application implementation, since it is up to the SDK user to define the ping semantics. This makes writing unit tests for custom pings a bit more involved.
+
+One possible strategy could be to wrap the Glean SDK API call to send the ping in a function that can be mocked in the unit test. This would allow for checking the status and the values of the metrics contained in the ping at the time in which the application would have sent it.
+
+## Example testing of a custom ping
+
+Let us start by defining a custom ping with a sample metric in it. Here is the `pings.yaml` file:
+
+```yaml
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
```

Comment:
I feel like the license header just adds clutter to the docs, but don't feel strongly about it...

Reply:
Good point :) Let me remove this